google/auth/aws.py

```diff
@@ -603,20 +610,37 @@ def __init__(
         self,
         audience,
         subject_token_type,
-        token_url,
+        token_url=external_account._DEFAULT_TOKEN_URL,
         credential_source=None,
+        aws_security_credentials_supplier=None,
         *args,
         **kwargs
     ):
         """Instantiates an AWS workload external account credentials object.
 
         Args:
             audience (str): The STS audience field.
-            subject_token_type (str): The subject token type.
-            token_url (str): The STS endpoint URL.
+            subject_token_type (str): The subject token type based on the Oauth2.0 token exchange spec.
+                Expected values include:
+                    “urn:ietf:params:aws:token-type:aws4_request”
+            token_url (str): The STS endpoint URL. If not provided, will default to "https://sts.googleapis.com/v1/token".
             credential_source (Mapping): The credential source dictionary used
                 to provide instructions on how to retrieve external credential
-                to be exchanged for Google access tokens.
+                to be exchanged for Google access tokens. Either a credential source or an AWS security
+                credentials supplier must be provided.
+                Example credential_source for AWS credential::
+
+                    {
+                        "environment_id": "aws1",
+                        "regional_cred_verification_url": "https://sts.{region}.amazonaws.com?Action=GetCallerIdentity&Version=2011-06-15",
+                        "region_url": "http://169.254.169.254/latest/meta-data/placement/availability-zone",
+                        "url": "http://169.254.169.254/latest/meta-data/iam/security-credentials",
+                        imdsv2_session_token_url": "http://169.254.169.254/latest/api/token"
+                    }
+            aws_security_credentials_supplier (AwsSecurityCredentialsSupplier): Optional AWS security credentials supplier.
```

Comment:
Should this be a union type?

Reply:
Also should be optional

Reply:
Misunderstood the doc string, but I suppose `credential_source` can be overloaded to use this type as well, to achieve the mutual exclusion.

Not a big deal, marking the type as Optional is enough to close this.